app/models/library_entry.rb

```diff
@@ -164,12 +167,22 @@ def kind
   end
 
   before_save do
-    if status_changed? && completed? && media&.progress_limit
-      # When marked completed, we try to update progress to the cap
-      self.progress = media.progress_limit
-    elsif !status_changed? && progress == media&.progress_limit
-      # When in current and progress equals total episodes
-      self.status = :completed
+    if status_changed? && completed?
+      # update progress to the cap
+      self.progress = media.progress_limit if media&.progress_limit
+      # set finished_at and started_at for the first consume
+      self.started_at ||= Time.now unless imported
+      self.finished_at ||= Time.now unless imported
```

Comment:
ah. hm. i wonder why i was getting new values on imported entries. 🤔  ill test it again, maybe i missed something.